controllers/machine_controller_phases.go

```diff
@@ -58,8 +58,8 @@ func (r *MachineReconciler) reconcilePhase(_ context.Context, m *clusterv1.Machi
 		m.Status.SetTypedPhase(clusterv1.MachinePhaseProvisioning)
 	}
 
-	// Set the phase to "provisioned" if there is a NodeRef.
-	if m.Status.NodeRef != nil {
+	// Set the phase to "provisioned" if there is a provider ID and the infrastructure is ready.
+	if m.Spec.ProviderID != nil && m.Status.InfrastructureReady {
 		m.Status.SetTypedPhase(clusterv1.MachinePhaseProvisioned)
```

Comment:
I think the idea in https://github.com/kubernetes-sigs/cluster-api/pull/1720 was that if the InfraMachine stops being ready, the Machine doesn't stay in Running... should this just be `if m.Spec.ProviderID != nil` ?

Reply:
If it flips? Yeah we I can remove the latter check, and just check on ProviderID, sgtm

Reply:
Actually reading down the comment thread in #1720 looks like there was some discussion around phases going backwards. What the expected phase now if a Machine has been created (provider ID is set) but InfraMachine ready is false?

Reply:
I think if InfraMachine ready status flips it goes back to Provisioning phase, since it has the providerId.
As mentioned in #1720, assumed that phases is a one way 
and if InfraMachine flips it will be accompanied by a failurereason / failuremessage, which will put machine in failure phase.

The provisioned phase is confusing on where it stands in the lifecycle of the machine.